frontend/src/components/match/MatchDialog.jsx

```diff
@@ -0,0 +1,107 @@
+import {
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  Button,
+  useDisclosure,
+} from '@chakra-ui/react'
+import { useState, useEffect } from 'react'
+import MatchForm from './MatchForm'
+import io from 'socket.io-client'
+import MatchTimer from './MatchTimer'
+import MatchTimeout from './MatchTimeout'
+import { useNavigate } from 'react-router-dom'
+import { URI_MATCHING_SERVICE, EVENT_EMIT, EVENT_LISTEN } from '../../constants'
+import MatchError from './MatchError'
+
+const phases = [
```

Comment:
Can consider making it an fake enum in js:
e.g. 
const Direction = {
  Up: 'Up',
  Down: 'Down',
  Left: 'Left',
  Right: 'Right'
};